packages/ts/react-crud/src/autoform.tsx

```diff
@@ -45,16 +45,81 @@ export type AutoFormLayoutProps = Readonly<{
 
 export type AutoFormProps<M extends AbstractModel = AbstractModel> = ComponentStyleProps &
   Readonly<{
+    /**
+     * The service to use for saving and deleting items. This must be a
+     * TypeScript service that has been generated by Hilla from a backend Java
+     * service that implements the `dev.hilla.crud.CrudService` interface.
+     */
     service: CrudService<Value<M>>;
+    /**
+     * The entity model to use, which determines which fields to show in the
+     * form. This must be a Typescript model class that has been generated by
+     * Hilla from a backend Java class. The model must match with the type of
+     * the items handled by the service. For example, a `PersonModel` can be
+     * used with a service that handles `Person` instances.
+     *
+     * By default, the form shows fields for all properties of the model which
+     * have a type that is supported. Use the `customLayoutRenderer` option to
+     * customize which fields to show and how to layout them.
+     */
     model: DetachedModelConstructor<M>;
+    /**
+     * The item to edit in the form. The form fields are automatically populated
+     * with values from the item's properties. In order to create a new item,
+     * either pass `null`, or leave this prop as undefined.
+     *
+     * Use the `afterSubmit` callback to get notified when the item has been
+     * saved.
+     */
     item?: Value<M> | typeof emptyItem | null;
+    /**
+     * Whether the form should be disabled. This disables all form fields and
+     * all buttons.
+     */
     disabled?: boolean;
+    /**
+     * Allows to customize the layout of the form, either by specifying rows of
+     * fields, or by providing a custom renderer. Please check the component's
+     * documentation for details.
+     */
     customLayoutRenderer?: AutoFormLayoutProps | ComponentType<AutoFormLayoutRendererProps<M>>;
```

Comment:
Maybe got a bit lazy here, but it seems difficult to succinctly explain how all these options work. The alternative was to convey what it does, and leave the how to the user documentation.